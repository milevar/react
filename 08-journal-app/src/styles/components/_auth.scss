
.auth_main {
  display: flex;
  background-color: $primary;
  align-items: center;
  justify-content: center;
  margin: 0px;
  height: 100vh;
  width: 100vw;
}

.auth_box-container {
  background-color: white;
  box-shadow: 0px 3px $dark-grey;
  border-radius: 2px;
  padding: 20px;
  width: 300px;
}

.auth_title {
  color: darken($color: $primary, $amount: 20);
  margin-bottom: 20px;
}

.auth_input {
  color: $dark-grey;
  border: 0px;
  border-bottom: 1px solid solid $light-grey;
  margin-bottom: 10px;
  height: 20px;
  width: 100%;

  transition: border-bottom .3s ease;
  &:focus {
    border-bottom: 1px solid $primary;
    outline: none;
  }
}

.auth_social-networks {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding-top: 20px;
  padding-bottom: 20px;
  width: 100%;
}

.auth_alert-error {
  background-color: red;
  border-radius: 5px;
  color: white;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}
