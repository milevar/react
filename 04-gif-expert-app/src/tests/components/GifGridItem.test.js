import React from "react";
import {shallow} from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem />', () => {

    const title = 'New Title';
    const url = "https://localhost/algo.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />)


    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener una img con url y alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener un animate_fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    })
})


