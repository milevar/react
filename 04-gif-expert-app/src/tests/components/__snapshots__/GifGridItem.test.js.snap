// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="New Title"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    New Title
     
  </p>
</div>
`;
